cddl_gen/cddl_gen.py

```diff
@@ -759,12 +777,19 @@ def post_validate(self):
         if self.cbor:
             self.cbor.post_validate()
 
+    def post_validate_control_group(self):
+        if self.type != "GROUP":
+            raise TypeError("control groups must be of GROUP type.")
+        for c in self.value:
+            if c.type != "UINT" or c.value is None or c.value < 0:
+                raise TypeError("control group members must be literal positive integers.")
+
     # Parses entire instr and returns a list of instances.
     def parse(self, instr):
         instr = instr.strip()
         values = []
         while instr != '':
-            value = type(self)(*self.init_args(), **self.init_kwargs())
+            value = type(self)(*self.init_args(), **self.init_kwargs(), base_stem=self.base_stem)
```

Comment:
Is this creating a class by building a constructor call with `self`'s type and a param tuple?

Reply:
It's instantiating a new object with some default values, as well as the same base_stem.